src/ExpressionEvaluator/Core/Source/ResultProvider/ResultProvider.cs

```diff
@@ -97,11 +103,11 @@ void IDkmClrResultProvider.GetItems(DkmEvaluationResultEnumContext enumContext,
                     return;
                 }
 
-                completionRoutine(GetItems(enumContext.InspectionContext, dataItem.EvalResultDataItem, startIndex, count));
+                completionRoutine(GetItems(dataItem.EvalResultDataItem, workList, startIndex, count, enumContext.InspectionContext));
             }
-            catch (Exception e) when (ExpressionEvaluatorFatalError.CrashIfFailFastEnabled(e))
+            catch (Exception e) when (ExpressionEvaluatorFatalError.ReportNonFatalException(e, DkmComponentManager.ReportCurrentNonFatalException))
             {
-                throw ExceptionUtilities.Unreachable;
+                completionRoutine(DkmEvaluationEnumAsyncResult.CreateErrorResult(e));
```

Comment:
If the exception is called from the "kick off" method (for example, IDkmClrResultProvider.GetItems), the Dispatcher should automatically call the completion routine for you.  Is that not working?  The thing to be careful of is throwing exceptions in your own continuations or methods called by those continuations.